app/src/main/java/org/wikipedia/feed/becauseyouread/BecauseYouReadClient.kt

```diff
@@ -29,7 +31,9 @@ class BecauseYouReadClient(
                 cb.success(emptyList())
             }
         ) {
-            val entries = AppDatabase.instance.historyEntryWithImageDao().findEntryForReadMore(age, context.resources.getInteger(R.integer.article_engagement_threshold_sec))
+            val entries = withContext(Dispatchers.IO) {
+                AppDatabase.instance.historyEntryWithImageDao().findEntryForReadMore(age, context.resources.getInteger(R.integer.article_engagement_threshold_sec))
```

Comment:
@cooltey A more comprehensive solution is to make the actual Dao function `suspend`, which automatically puts it in an IO dispatcher. See #5419

Reply:
It looks good to me! That was my original approach but turned into this one in the end.